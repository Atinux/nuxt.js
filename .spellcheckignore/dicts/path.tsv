basedir
basename
dirname
extname
filename
sep