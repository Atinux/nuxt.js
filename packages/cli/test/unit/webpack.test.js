import path from 'path'
import consola from 'consola'
import prettyFormat from 'pretty-format'
import { NuxtCommand, getWebpackConfig } from '../..'
import webpackCommand from '../../src/commands/webpack'

const replaceAll = (str, a, b) => str.split(a).join(b)
const nuxtDir = path.join(__dirname, '../../../..')
const maskDir = str => replaceAll(str, nuxtDir, '<nuxtDir>')

const tests = [
  'devtool',
  'resolve alias',
  'module.rules',
  'module.rules test=.jsx',
  'module.rules loader=vue-',
  'module.rules loader=.*-loader'
]

describe('webpack', () => {
  beforeAll(() => {
    process.stdout.isTTY = false
  })

  afterEach(() => {
    jest.resetAllMocks()
  })

  test('getWebpackConfig()', async () => {
    const webpackConfig = await getWebpackConfig('Client')
    expect(maskDir(prettyFormat(webpackConfig.module.rules[0]))).toMatchSnapshot()
  })

  test('nuxt webpack no match', async () => {
    const cmd = NuxtCommand.from(webpackCommand, ['module.rules', 'loader=foobar'])
    process.exit = jest.fn()
    await expect(cmd.run()).rejects
      .toThrow('No match in webpack config for path:module.rules query:loader=foobar. Try inspecting without query and path.')
  })

  for (const testCase of tests) {
    test('nuxt webpack ' + testCase, async () => {
      const cmd = NuxtCommand.from(webpackCommand, testCase.split(' '))
      await cmd.run()
      expect(maskDir(consola.log.mock.calls[0][0])).toMatchSnapshot()
    })
  }
})
