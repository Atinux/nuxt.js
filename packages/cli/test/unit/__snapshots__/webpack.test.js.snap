// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack getWebpackConfig() 1`] = `
"Object {
  \\"loader\\": \\"vue-loader\\",
  \\"options\\": Object {
    \\"productionMode\\": true,
    \\"transformAssetUrls\\": Object {
      \\"embed\\": \\"src\\",
      \\"object\\": \\"src\\",
      \\"source\\": \\"src\\",
      \\"video\\": \\"src\\",
    },
  },
  \\"test\\": /\\\\.vue$/i,
}"
`;

exports[`webpack nuxt webpack devtool 1`] = `
"false
"
`;

exports[`webpack nuxt webpack module.rules 1`] = `
"[
  {
    test: /\\\\.vue$/i,
    loader: 'vue-loader',
    options: {
      transformAssetUrls: { video: 'src', source: 'src', object: 'src', embed: 'src' },
      productionMode: true
    }
  },
  {
    test: /\\\\.pug$/i,
    oneOf: [
      {
        resourceQuery: /^\\\\?vue/i,
        use: [ { loader: 'pug-plain-loader', options: {} } ]
      },
      {
        use: [ 'raw-loader', { loader: 'pug-plain-loader', options: {} } ]
      }
    ]
  },
  {
    test: /\\\\.jsx?$/i,
    exclude: [Function: exclude],
    use: [
      {
        loader: '<nuxtDir>/node_modules/babel-loader/lib/index.js',
        options: {
          configFile: false,
          babelrc: false,
          cacheDirectory: false,
          envName: 'client',
          presets: [
            [
              '<nuxtDir>/packages/babel-preset-app/src/index.js',
              {}
            ]
          ]
        }
      }
    ]
  },
  {
    test: /\\\\.css$/i,
    oneOf: [
      {
        resourceQuery: /module/,
        use: [
          { loader: 'vue-style-loader', options: { sourceMap: false } },
          {
            loader: 'css-loader',
            options: {
              modules: { localIdentName: '[local]_[hash:base64:5]' },
              sourceMap: false,
              importLoaders: 2,
              onlyLocals: false
            }
          },
          {
            loader: 'postcss-loader',
            options: {
              sourceMap: false,
              plugins: [
                [Function] {
                  postcssPlugin: 'postcss-import',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-url',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-preset-env',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'cssnano',
                  postcssVersion: '7.0.27'
                }
              ],
              order: 'presetEnvAndCssnanoLast'
            }
          }
        ]
      },
      {
        use: [
          { loader: 'vue-style-loader', options: { sourceMap: false } },
          {
            loader: 'css-loader',
            options: { sourceMap: false, importLoaders: 2, onlyLocals: false }
          },
          {
            loader: 'postcss-loader',
            options: {
              sourceMap: false,
              plugins: [
                [Function] {
                  postcssPlugin: 'postcss-import',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-url',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-preset-env',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'cssnano',
                  postcssVersion: '7.0.27'
                }
              ],
              order: 'presetEnvAndCssnanoLast'
            }
          }
        ]
      }
    ]
  },
  {
    test: /\\\\.p(ost)?css$/i,
    oneOf: [
      {
        resourceQuery: /module/,
        use: [
          { loader: 'vue-style-loader', options: { sourceMap: false } },
          {
            loader: 'css-loader',
            options: {
              modules: { localIdentName: '[local]_[hash:base64:5]' },
              sourceMap: false,
              importLoaders: 2,
              onlyLocals: false
            }
          },
          {
            loader: 'postcss-loader',
            options: {
              sourceMap: false,
              plugins: [
                [Function] {
                  postcssPlugin: 'postcss-import',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-url',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-preset-env',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'cssnano',
                  postcssVersion: '7.0.27'
                }
              ],
              order: 'presetEnvAndCssnanoLast'
            }
          }
        ]
      },
      {
        use: [
          { loader: 'vue-style-loader', options: { sourceMap: false } },
          {
            loader: 'css-loader',
            options: { sourceMap: false, importLoaders: 2, onlyLocals: false }
          },
          {
            loader: 'postcss-loader',
            options: {
              sourceMap: false,
              plugins: [
                [Function] {
                  postcssPlugin: 'postcss-import',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-url',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-preset-env',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'cssnano',
                  postcssVersion: '7.0.27'
                }
              ],
              order: 'presetEnvAndCssnanoLast'
            }
          }
        ]
      }
    ]
  },
  {
    test: /\\\\.less$/i,
    oneOf: [
      {
        resourceQuery: /module/,
        use: [
          { loader: 'vue-style-loader', options: { sourceMap: false } },
          {
            loader: 'css-loader',
            options: {
              modules: { localIdentName: '[local]_[hash:base64:5]' },
              sourceMap: false,
              importLoaders: 2,
              onlyLocals: false
            }
          },
          {
            loader: 'postcss-loader',
            options: {
              sourceMap: false,
              plugins: [
                [Function] {
                  postcssPlugin: 'postcss-import',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-url',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-preset-env',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'cssnano',
                  postcssVersion: '7.0.27'
                }
              ],
              order: 'presetEnvAndCssnanoLast'
            }
          },
          { loader: 'less-loader', options: { sourceMap: false } }
        ]
      },
      {
        use: [
          { loader: 'vue-style-loader', options: { sourceMap: false } },
          {
            loader: 'css-loader',
            options: { sourceMap: false, importLoaders: 2, onlyLocals: false }
          },
          {
            loader: 'postcss-loader',
            options: {
              sourceMap: false,
              plugins: [
                [Function] {
                  postcssPlugin: 'postcss-import',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-url',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-preset-env',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'cssnano',
                  postcssVersion: '7.0.27'
                }
              ],
              order: 'presetEnvAndCssnanoLast'
            }
          },
          { loader: 'less-loader', options: { sourceMap: false } }
        ]
      }
    ]
  },
  {
    test: /\\\\.sass$/i,
    oneOf: [
      {
        resourceQuery: /module/,
        use: [
          { loader: 'vue-style-loader', options: { sourceMap: false } },
          {
            loader: 'css-loader',
            options: {
              modules: { localIdentName: '[local]_[hash:base64:5]' },
              sourceMap: false,
              importLoaders: 2,
              onlyLocals: false
            }
          },
          {
            loader: 'postcss-loader',
            options: {
              sourceMap: false,
              plugins: [
                [Function] {
                  postcssPlugin: 'postcss-import',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-url',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-preset-env',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'cssnano',
                  postcssVersion: '7.0.27'
                }
              ],
              order: 'presetEnvAndCssnanoLast'
            }
          },
          {
            loader: 'sass-loader',
            options: { sassOptions: { indentedSyntax: true }, sourceMap: false }
          }
        ]
      },
      {
        use: [
          { loader: 'vue-style-loader', options: { sourceMap: false } },
          {
            loader: 'css-loader',
            options: { sourceMap: false, importLoaders: 2, onlyLocals: false }
          },
          {
            loader: 'postcss-loader',
            options: {
              sourceMap: false,
              plugins: [
                [Function] {
                  postcssPlugin: 'postcss-import',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-url',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-preset-env',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'cssnano',
                  postcssVersion: '7.0.27'
                }
              ],
              order: 'presetEnvAndCssnanoLast'
            }
          },
          {
            loader: 'sass-loader',
            options: { sassOptions: { indentedSyntax: true }, sourceMap: false }
          }
        ]
      }
    ]
  },
  {
    test: /\\\\.scss$/i,
    oneOf: [
      {
        resourceQuery: /module/,
        use: [
          { loader: 'vue-style-loader', options: { sourceMap: false } },
          {
            loader: 'css-loader',
            options: {
              modules: { localIdentName: '[local]_[hash:base64:5]' },
              sourceMap: false,
              importLoaders: 2,
              onlyLocals: false
            }
          },
          {
            loader: 'postcss-loader',
            options: {
              sourceMap: false,
              plugins: [
                [Function] {
                  postcssPlugin: 'postcss-import',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-url',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-preset-env',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'cssnano',
                  postcssVersion: '7.0.27'
                }
              ],
              order: 'presetEnvAndCssnanoLast'
            }
          },
          { loader: 'sass-loader', options: { sourceMap: false } }
        ]
      },
      {
        use: [
          { loader: 'vue-style-loader', options: { sourceMap: false } },
          {
            loader: 'css-loader',
            options: { sourceMap: false, importLoaders: 2, onlyLocals: false }
          },
          {
            loader: 'postcss-loader',
            options: {
              sourceMap: false,
              plugins: [
                [Function] {
                  postcssPlugin: 'postcss-import',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-url',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-preset-env',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'cssnano',
                  postcssVersion: '7.0.27'
                }
              ],
              order: 'presetEnvAndCssnanoLast'
            }
          },
          { loader: 'sass-loader', options: { sourceMap: false } }
        ]
      }
    ]
  },
  {
    test: /\\\\.styl(us)?$/i,
    oneOf: [
      {
        resourceQuery: /module/,
        use: [
          { loader: 'vue-style-loader', options: { sourceMap: false } },
          {
            loader: 'css-loader',
            options: {
              modules: { localIdentName: '[local]_[hash:base64:5]' },
              sourceMap: false,
              importLoaders: 2,
              onlyLocals: false
            }
          },
          {
            loader: 'postcss-loader',
            options: {
              sourceMap: false,
              plugins: [
                [Function] {
                  postcssPlugin: 'postcss-import',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-url',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-preset-env',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'cssnano',
                  postcssVersion: '7.0.27'
                }
              ],
              order: 'presetEnvAndCssnanoLast'
            }
          },
          { loader: 'stylus-loader', options: { sourceMap: false } }
        ]
      },
      {
        use: [
          { loader: 'vue-style-loader', options: { sourceMap: false } },
          {
            loader: 'css-loader',
            options: { sourceMap: false, importLoaders: 2, onlyLocals: false }
          },
          {
            loader: 'postcss-loader',
            options: {
              sourceMap: false,
              plugins: [
                [Function] {
                  postcssPlugin: 'postcss-import',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-url',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'postcss-preset-env',
                  postcssVersion: '7.0.27'
                },
                [Function] {
                  postcssPlugin: 'cssnano',
                  postcssVersion: '7.0.27'
                }
              ],
              order: 'presetEnvAndCssnanoLast'
            }
          },
          { loader: 'stylus-loader', options: { sourceMap: false } }
        ]
      }
    ]
  },
  {
    test: /\\\\.(png|jpe?g|gif|svg|webp)$/i,
    use: [
      {
        loader: 'url-loader',
        options: { limit: 1000, name: 'img/[contenthash:7].[ext]' }
      }
    ]
  },
  {
    test: /\\\\.(woff2?|eot|ttf|otf)(\\\\?.*)?$/i,
    use: [
      {
        loader: 'url-loader',
        options: { limit: 1000, name: 'fonts/[contenthash:7].[ext]' }
      }
    ]
  },
  {
    test: /\\\\.(webm|mp4|ogv)$/i,
    use: [
      {
        loader: 'file-loader',
        options: { name: 'videos/[contenthash:7].[ext]' }
      }
    ]
  }
]
"
`;

exports[`webpack nuxt webpack module.rules loader=.*-loader 1`] = `
"{
  test: /\\\\.vue$/i,
  loader: 'vue-loader',
  options: {
    transformAssetUrls: { video: 'src', source: 'src', object: 'src', embed: 'src' },
    productionMode: true
  }
}
"
`;

exports[`webpack nuxt webpack module.rules loader=vue- 1`] = `
"{
  test: /\\\\.vue$/i,
  loader: 'vue-loader',
  options: {
    transformAssetUrls: { video: 'src', source: 'src', object: 'src', embed: 'src' },
    productionMode: true
  }
}
"
`;

exports[`webpack nuxt webpack module.rules test=.jsx 1`] = `
"{
  test: /\\\\.jsx?$/i,
  exclude: [Function: exclude],
  use: [
    {
      loader: '<nuxtDir>/node_modules/babel-loader/lib/index.js',
      options: {
        configFile: false,
        babelrc: false,
        cacheDirectory: false,
        envName: 'client',
        presets: [
          [
            '<nuxtDir>/packages/babel-preset-app/src/index.js',
            {}
          ]
        ]
      }
    }
  ]
}
"
`;

exports[`webpack nuxt webpack resolve alias 1`] = `
"{
  '~~': '<nuxtDir>/resolve',
  '@@': '<nuxtDir>/resolve',
  '~': '<nuxtDir>/resolve',
  '@': '<nuxtDir>/resolve',
  assets: '<nuxtDir>/resolve/assets',
  static: '<nuxtDir>/resolve/static',
  'vue-meta': '<nuxtDir>/node_modules/vue-meta/dist/vue-meta.esm.browser.js'
}
"
`;
